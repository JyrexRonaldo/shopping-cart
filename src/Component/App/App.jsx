import { Link, Outlet } from "react-router-dom";
import styles from "./App.module.css";
import { useEffect, useState } from "react";
import StoreDataContext from "../StoreDataContext/StoreDataContext";

function App() {
  const useProductsData = () => {
    const [productsData, setProductsData] = useState(null);
    const [error, setError] = useState(null);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
      fetch("https://fakestoreapi.com/products")
        .then((response) => {
          if (response.status >= 400) {
            throw new Error("server error");
          }
          return response.json();
        })
        .then((response) => setProductsData(response))
        .catch((error) => setError(error))
        .finally(() => setLoading(false));
    }, []);

    return { productsData, error, loading };
  };

  const [cartItems, setCartItems] = useState({});

  let totalCartItems = 0;

  for (const key in cartItems) {
    if (Object.prototype.hasOwnProperty.call(cartItems, key)) {
      totalCartItems += +cartItems[key];
    }
  }

  const outletData = { ...useProductsData(), cartItems, setCartItems };

  return (
    <div className={styles.app}>
      <nav className={styles.navBar}>
        <Link to="/">
          <p>Suburban</p>
        </Link>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/shop">Shop</Link>
          </li>
          <li>
            <Link to="/cart">Cart {`(${totalCartItems})`}</Link>
          </li>
        </ul>
      </nav>

      <StoreDataContext.Provider value={outletData}>
        <Outlet />
      </StoreDataContext.Provider>
    </div>
  );
}

export default App;
