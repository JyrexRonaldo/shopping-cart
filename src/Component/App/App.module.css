.app {
  background-image: url(/background.jpg);
  background-size: cover;
  height: 100vh;
  display: flex;
  flex-direction: column;
}

.navBar {
  background-color: rgb(0, 0, 0);
  color: white;
  height: 8vh;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 30px;
}

.navBar ul {
  display: flex;
  gap: 15px;
}

.navBar a {
  color: white;
}
