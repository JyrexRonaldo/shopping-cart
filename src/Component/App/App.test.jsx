import App from "./App";
import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("App component", () => {
  it("it displays home page items", () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
