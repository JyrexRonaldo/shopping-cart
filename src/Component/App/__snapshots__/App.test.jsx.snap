// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > it displays home page items 1`] = `
<div>
  <div
    class="_app_d1d56a"
  >
    <nav
      class="_navBar_d1d56a"
    >
      <a
        href="/"
      >
        <p>
          Suburban
        </p>
      </a>
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/shop"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            href="/cart"
          >
            Cart 
            (0)
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
